app/services/EmploymentSessionService.scala

```diff
@@ -0,0 +1,143 @@
+ */
+
+package services
+
+import config.{AppConfig, ErrorHandler}
+import connectors.IncomeTaxUserDataConnector
+import connectors.httpParsers.IncomeTaxUserDataHttpParser.IncomeTaxUserDataResponse
+import javax.inject.{Inject, Singleton}
+import models.{IncomeTaxUserData, User}
+import models.employment.{AllEmploymentData, EmploymentExpenses, EmploymentSource}
+import models.mongo.{EmploymentCYAModel, EmploymentUserData}
+import org.joda.time.{DateTime, DateTimeZone}
+import play.api.Logging
+import play.api.i18n.MessagesApi
+import play.api.mvc.Results.Redirect
+import play.api.mvc.{Request, Result}
+import repositories.EmploymentUserDataRepository
+import uk.gov.hmrc.http.HeaderCarrier
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class EmploymentSessionService @Inject()(employmentUserDataRepository: EmploymentUserDataRepository,
+                                         incomeTaxUserDataConnector: IncomeTaxUserDataConnector,
+                                         implicit private val appConfig: AppConfig,
+                                         val messagesApi: MessagesApi,
+                                         errorHandler: ErrorHandler) extends Logging {
+
+  def findPreviousEmploymentUserData(user: User[_], taxYear: Int)(result: AllEmploymentData => Result)
+                                    (implicit request: Request[_], hc: HeaderCarrier, ec: ExecutionContext): Future[Result] = {
+
+    getPriorData(taxYear)(user,hc).map {
+      case Right(IncomeTaxUserData(Some(employmentData))) => result(employmentData)
+      case Right(IncomeTaxUserData(None)) =>
+        logger.info(s"[IncomeTaxUserDataService][findUserData] No employment data found for user. SessionId: ${user.sessionId}")
+        Redirect(appConfig.incomeTaxSubmissionOverviewUrl(taxYear))
+      case Left(error) => errorHandler.handleError(error.status)
+    }
+  }
+
+  private def getPriorData(taxYear: Int)(implicit user: User[_], hc: HeaderCarrier): Future[IncomeTaxUserDataResponse] = {
+    incomeTaxUserDataConnector.getUserData(user.nino, taxYear)(hc.withExtraHeaders("mtditid" -> user.mtditid))
+  }
+
+  def getSessionData(taxYear: Int, employmentId: String)(implicit user: User[_]): Future[Option[EmploymentUserData]] = {
+    employmentUserDataRepository.find(taxYear, employmentId)
+  }
+
+  //scalastyle:off
+  def updateSessionData[A](employmentId: String, cyaModel: EmploymentCYAModel, taxYear: Int,
+                           needsCreating: Boolean, isPriorSubmission: Boolean)(onFail: A)(onSuccess: A)
+                          (implicit user: User[_], ec: ExecutionContext): Future[A] = {
+
+    val userData = EmploymentUserData(
+      user.sessionId,
+      user.mtditid,
+      user.nino,
+      taxYear,
+      employmentId,
+      isPriorSubmission = isPriorSubmission,
+      cyaModel,
+      DateTime.now(DateTimeZone.UTC)
+    )
+
+    //TODO See if we can remove the create / update and have one method
```

Comment:
Another TODO

Reply:
will be removed once Abdullah has done some investigation